code/modules/industrial_lift/elevator/elevator_machinery.dm

```diff
@@ -373,3 +389,385 @@
 		. += emissive_appearance(icon, light_mask, src, alpha = alpha)
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/elevator_control_panel, 31)
+
+/obj/item/assembly/control/elevator
+	name = "elevator controller"
+	desc = "A small device used to call elevators to the current floor."
+	/// A weakref to the lift_master datum we control
+	var/datum/weakref/lift_weakref
+
+/obj/item/assembly/control/elevator/Initialize(mapload)
+	. = ..()
+
+	if(mapload)
+		return INITIALIZE_HINT_LATELOAD
+
+	var/datum/lift_master/lift = get_lift()
+	if(!lift)
+		return
+
+	lift_weakref = WEAKREF(lift)
+
+/obj/item/assembly/control/elevator/LateInitialize()
+	var/datum/lift_master/lift = get_lift()
+	if(!lift)
+		log_mapping("Elevator call button at [AREACOORD(src)] found no associated lift to link with, this may be a mapping error.")
+		return
+
+	lift_weakref = WEAKREF(lift)
+
+// Emagging elevator buttons will disable safeties
+/obj/item/assembly/control/elevator/emag_act(mob/user, obj/item/card/emag/emag_card)
+	if(obj_flags & EMAGGED)
+		return
+
+	var/datum/lift_master/lift = lift_weakref?.resolve()
+	if(!lift)
+		return
+
+	for(var/obj/structure/industrial_lift/lift_platform as anything in lift.lift_platforms)
+		lift_platform.violent_landing = TRUE
+		lift_platform.warns_on_down_movement = FALSE
+		lift_platform.elevator_vertical_speed = initial(lift_platform.elevator_vertical_speed) * 0.5
+
+	// Note that we can either be emagged by having the button we are inside swiped,
+	// or by someone emagging the assembly directly after removing it (to be cheeky)
+	var/atom/balloon_alert_loc = get(src, /obj/machinery/button) || src
+	balloon_alert_loc.balloon_alert(user, "safeties overridden")
+	obj_flags |= EMAGGED
+	return TRUE
```

Comment:
should the other returns in this proc return `FALSE` then, instead of null?

Reply:
I don't think a return status is needed at all.